File: Java/프로그래머스/3/152995. 인사고과/인사고과.java
import java.util.*;
class Solution {
    public int solution(int[][] scores) {
        int answer = 1;
        int a = scores[0][0];
        int b = scores[0][1];
        int sum = a + b;
        Arrays.sort(scores, (o1, o2) -> o1[0] == o2[0] ? o1[1] - o2[1] : o2[0] - o1[0]);

        int temp = 0;
        for (int[] score : scores) {
            if (a < score[0] && b < score[1]) {
                return -1;
            }

            if (temp <= score[1]) {
                temp = score[1];
                if (score[0] + score[1] > sum) {
                    answer++;
                }
            }
        }
        return answer;
    }
}

File: Python/프로그래머스/3/152995. 인사고과/인사고과.py
def solution(scores):
    answer = 1
    a, b = scores[0]
    total = a + b

    scores = sorted(scores, key=lambda x: (-x[0], x[1]))

    tmp = 0
    for x, y in scores:
        if a < x and b < y:
            return -1

        if tmp <= y:
            tmp = y
            if x + y > total:
                answer += 1

    return answer